KarhooUISDK/Screens/BookingScreen/QuoteCategoryBarMVP/KarhooQuoteCategoryBarPresenter.swift

```diff
@@ -48,6 +48,14 @@ final class KarhooQuoteCategoryBarPresenter: QuoteCategoryBarPresenter {
         guard self.categories != categories else {
             return
         }
+        
+        // Reset the selected index if the old category count doesn't match the new one
+        // This fixes the scenario where the user has the "All" category selected on ASAP, then chooses to pre-book,
+        // in which case more or less categories may appear, offsetting the initial selection
+        if self.categories.count != categories.count {
+            self.selectedIndex = nil
```

Comment:
Shouldn't we try to compare previous categories with new once and if it's possible keep selected filter type? And if no possible (previously selected category is missing) then reset selection?

Reply:
Sorry I missed your comment. It would be an improvement, yes. My fix is a bandaid for a specific scenario where more or less categories are received. This normally only happens when you switch between asap and pre-book or between pick up and drop-off. The rest of the time the component will have the same behavior as before which should be good enough until we move on to the new quote list implementation.